### DISCUSSION TITLE
Quarkus Qute – A Test Ride - Gunnar Morling

### DISCUSSION THREAD

**Discussion by blog-comment-importer** [gunnarmorling/discussions.morling.dev#269] [Announcements]
https://www.morling.dev/blog/quarkus-qute-test-ride/

**Comment by blog-comment-importer** [comment]
_**dasniko** wrote at 2020-01-10T14:03:09Z (comment imported from Disqus):_

<p>Nice, thanks for that, I also played with Quarkus and Qute a bit.</p><p>I have this challenge, perhaps you know a solution:<br>The bean I use in my template contains a <code>java.util.Map</code> and I need to test, if a certain value is contained in this map. I thought I'd work this way, but won't:</p>

```
<div class="{#if reg.validationErrors.containsKey('name')}has-error{/if}">
```

Instead, I'm using this as a workaround. Works, but ugly:

```
<div class="form-group {#each reg.validationErrors}{#if it.key == 'name'}has-error{/if}{/each}">
```

<p>Do you know, if invoking methods of a map work with Qute? Or not yet? Is there any other (nicer) workaround?<br>I know, I could change my data for the template, but that would require to change some more backend logic (it's an existing app I'll try to migrate).</p>

**Reply by blog-comment-importer** [reply]
_**Gunnar Morling** wrote at 2020-01-10T14:16:26Z (comment imported from Disqus):_

<p>Yes, method invocation support in templates still is a bit sparse at this point. There's issue <a href="https://github.com/quarkusio/quarkus/issues/6369" rel="nofollow noopener" title="https://github.com/quarkusio/quarkus/issues/6369">#6369</a> for improving that. For the time being your work-around looks good; perhaps you could also use an <a href="https://quarkus.io/guides/qute#parameter-declarations-and-template-extension-methods" rel="nofollow noopener" title="https://quarkus.io/guides/qute#parameter-declarations-and-template-extension-methods">extension method</a>, but I'm not quite sure whether they support parameters already (as you'd need it here to invoke reg.hasValidationError('name') from within your template).</p>

**Reply by blog-comment-importer** [reply]
_**dasniko** wrote at 2020-01-10T14:24:20Z (comment imported from Disqus):_

<p>Thanks. I already feared, that I'll have to wait until one of the next releases... Let's see what's comming.</p>

**Reply by blog-comment-importer** [reply]
_**mkouba** wrote at 2020-01-13T10:03:44Z (comment imported from Disqus):_

<p>This is a bug. See <a href="https://github.com/quarkusio/quarkus/issues/6519" rel="nofollow noopener" title="https://github.com/quarkusio/quarkus/issues/6519">https://github.com/quarkusi...</a>. Note that Qute is very extensible so there is usually a workaround ;-).</p>

**Comment by blog-comment-importer** [comment]
_**Vincent Leon** wrote at 2020-01-19T16:29:42Z (comment imported from Disqus):_

<p>Thanks for this great article about Qute (that I never heard of before), a good opportunity to continue my journey through all the Quarkus features!</p>

**Reply by blog-comment-importer** [reply]
_**Gunnar Morling** wrote at 2020-01-19T17:00:04Z (comment imported from Disqus):_

<p>My pleasure; very happy to hear you found the post useful.</p>

**Comment by blog-comment-importer** [comment]
_**Hossein Mahdavipour** wrote at 2020-05-17T07:34:59Z (comment imported from Disqus):_

<p>wow. excellent article.<br>thank you</p>

**Reply by blog-comment-importer** [reply]
_**Gunnar Morling** wrote at 2020-05-17T11:10:42Z (comment imported from Disqus):_

<p>Thanks, that's much appreciated!</p>

**Comment by blog-comment-importer** [comment]
_**Taysay** wrote at 2020-07-05T21:16:44Z (comment imported from Disqus):_

<p>This is awesome, thank you very much.</p>

**Comment by blog-comment-importer** [comment]
_**Nhlanhla Mkhize** wrote at 2020-08-24T13:57:38Z (comment imported from Disqus):_

<p>Thank you for this article</p>

**Comment by Serkan80** [comment]
There is now a Quarkus Webframework based upon Qute called Quarkus Renarde, see here:

https://quarkiverse.github.io/quarkiverse-docs/quarkus-renarde/dev/index.html
